### PR TITLE
deprecates POST v1/nodes endpoint

### LINKED ISSUE #2099: machineDeployment "Ready to be shipped" tracking issue
This epos aims to track all remaining tasks until the machineDeployment feature is ready to be delivered in a release. Each task listed here should get an individual issue that tracks its details.

*API*

* [x] Add endpoints that manage machineDeployments
  * [x] #2212 creating `MachineDeployment`
  * [x] #2453 Update a `machineDeployment`
  * [x] #2213 listing `MachineDeployment`s
  * [x] #2214 getting a `MachineDeployment`
  * [x] #2215 deleting a `MachineDeployment`
* [x] Change the existing node endpoints to now show machines/nodes that are owned by a machineSet (#2216)
* [x] Change the existing node endpoints to only allow GET and DELETE and return a helpful error if other operations are attempted (#2217)

*UI*

* [x] Show machineDeployments
* [x] Discuss and implement how a machineDeployments status can be displayed
  * [ ]   Do we want to show individual machineSets and their configuration?
* [x] Allow updating machineDeployments
* [x] Continue to show machines and allow deleting them, but not creation of new ones
  * [x] Maybe implement this in a backwards-compatible way by checking if the API exposes the machineDeployment endpoint?

*Controllers*

* [x] Add mutatingAdmissionWebhok to machine-controller: kubermatic/machine-controller#227
* [x] Add the defaulting and validation from the upstream cluster-api project to that webhook
* [x] Move the existing defaulting and validation out of the machine-controller and into that webhook
* [x] Do the existing defaulting and validation at the highest possible level and only once (e.G. do it at machineDeployment level and not do it again for each individual machine)
* [x] Make the machine-controller drain nodes before it deletes machines: kubermatic/kubermatic#2158

### LINKED ISSUE #2217: Change existing Node/Machine API endpoints to refuse creating or modifying Machines
### !!! This story will remove stand-alone `Machine` support. Do not implement before full `MachineDeployment` support is implemented

**User Story**
The support for standalone `Machine` objects without `MachineDeployments` will be slowly phased out. Existing machines will be kept running, and the API will still allow getting and deleting them. However, no new standalone `Machine` objects can be allowed to be created, nor can an existing `Machine` be allowed to be modified.

**Acceptance criteria**
 * `POST` `/api/v1/projects/{project_id}/dc/{dc}/clusters/{cluster_id}/nodes` is no-op and returns a helpful error message to the caller.
 * There is a test that verifies the error message is being returned.
 * There is a test that verifies no changes are actually made.
 * Both adding a new node, as well as changing an existing node are tested.

### PR DESCRIPTION
**Pull request by p0lyn0mial**
**What this PR does / why we need it**: creating new nodes is deprecated, the users should create a `Node Deploymen` instead, using `POST /api/v1/projects/{project_id}/dc/{dc}/clusters/{cluster_id}/nodedeployments`

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2217 
Fixes #2099

**Special notes for your reviewer**: I slightly extended the error msg returned from API to contain additional msg helpful for users that don't use dashboard. The additional message will not be displayed by dashboard. 

`{"error":{"code":400,"message":"Creating Nodes is depreciated. Please create a NodeDeployment instead","details":"If you are calling this API endpoint directly then use POST \"v1/projects/{project_id}/dc/{dc}/clusters/{cluster_id}/nodedeployments\" instead"}}` 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by p0lyn0mial** [comment]
i'm going to add an issue to our backlog for removing depreciated endpoint completely from our API.

**Comment by maciaszczykm** [comment]
I think it also closes https://github.com/kubermatic/kubermatic/issues/2099. @alvaroaleman WDYT?

File: api/pkg/handler/node_v1.go [hunk 1]
```diff
@@ -215,84 +214,9 @@ func getNodeForCluster(projectProvider provider.ProjectProvider) endpoint.Endpoi
...
-		}
-
-		// Send machine resource to k8s
-		machine, err = machineClient.ClusterV1alpha1().Machines(machine.Namespace).Create(machine)
-		if err != nil {
-			return nil, fmt.Errorf("failed to create machine: %v", err)
-		}
-
-		return outputMachine(machine, nil, false)
+		return nil, k8cerrors.NewWithDetails(http.StatusBadRequest, "Creating Nodes is depreciated. Please create a Node Deployment instead", "If you are calling this API endpoint directrly then use POST \"v1/projects/{project_id}/dc/{dc}/clusters/{cluster_id}/nodedeployments\" instead")
```
Comment (maciaszczykm) [inline api/pkg/handler/node_v1.go]: s/depreciated/deprecated

Comment (maciaszczykm) [reply api/pkg/handler/node_v1.go]: Also in other places.

**Comment by p0lyn0mial** [comment]
> i'm going to add an issue to our backlog for removing depreciated endpoint completely from our API.

https://app.zenhub.com/workspaces/all-kubermatic-58863654794153994b62af76/issues/kubermatic/kubermatic/2645

**Pull request merged**